### ISSUE TITLE
Detelecine and Frame Rate selection

### ISSUE THREAD

**Issue by cameronks** [donmelton/video_transcoding#292]
When using Detelecine, is 23.976 always the correct choice to force as frame rate? I know `transcode-video` always forces 23.976 for DVD material. 

Would it not be safer to encode with `--limit-rate` as is the default in Handbrake presets? Wouldn't most modern players handle variable frame rate files?

Also, when forcing 23.976, files always show up with a slight variance due to the way Handbrake calculates this. Is this creating any additional problems or negating the choice of forcing this frame rate in the first place?

> Frame rate                               : 23.976 (24000/1001) FPS
Minimum frame rate                       : 23.974 FPS
Maximum frame rate                       : 23.981 FPS

**Comment by donmelton** [comment]
@cameronks Usually, yes, you want to use `23.976`. However are some _very_ rare cases with badly mastered BBC content where you want to use `25` as the argument to `--force-rate`.

And you definitely want to force instead of limit the frame rate. Otherwise `--filter detelecine` may not work correctly.

Just ignore the slight variance in frame rate from HandBrake. It won't affect playback at all. HandBrake is weird. What can I say. :)

**Comment by cameronks** [comment]
Thanks Don.

Here's one of the weirder source files that got me thinking about this. The source itself is variable. Even after detelecine, there is some combing so I'm using 

`--filter comb-detect --filter decomb --filter detelecine `

on these. Does that sounds about right?

> Video
ID                                       : 1
ID in the original source medium         : 224 (0xE0)
Format                                   : MPEG Video
Format version                           : Version 2
Format profile                           : Main@Main
Format settings                          : CustomMatrix / BVOP
Format settings, BVOP                    : Yes
Format settings, Matrix                  : Custom
Format settings, GOP                     : M=3, N=15
Format settings, picture structure       : Frame
Codec ID                                 : V_MPEG2
Codec ID/Info                            : MPEG 1 or 2 Video
Duration                                 : 10 min 24 s
Bit rate mode                            : Variable
Bit rate                                 : 4 722 kb/s
Maximum bit rate                         : 8 000 kb/s
Width                                    : 720 pixels
Height                                   : 480 pixels
Display aspect ratio                     : 4:3
Frame rate mode                          : Variable
Frame rate                               : 26.706 FPS
Original frame rate                      : 29.970 (30000/1001) FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Scan type                                : Interlaced
Scan order                               : Top Field First
Compression mode                         : Lossy
Bits/(Pixel*Frame)                       : 0.512
Time code of first frame                 : 00:59:58;00
Time code source                         : Group of pictures header
GOP, Open/Closed                         : Closed
Stream size                              : 351 MiB (91%)
Language                                 : English
Default                                  : No
Forced                                   : No
Original source medium                   : DVD-Video

**Comment by donmelton** [comment]
@cameronks Woof! Yeah, that _IS_ weird. :)

No idea where that's the right set of options but, hey, if it works and it doesn't look like ass, go for it!

**Comment by cameronks** [comment]
Ok. Im testing to see if 'same as source' works any better directly in Handbrake for these "special" files.

**Comment by cameronks** [comment]
I found more files like the one above. Actually a lot as this is how most of the episodes from The Simpsons look in MediaInfo:

> Video
ID                                       : 1
ID in the original source medium         : 224 (0xE0)
Format                                   : MPEG Video
Format version                           : Version 2
Format profile                           : Main@Main
Format settings                          : CustomMatrix / BVOP
Format settings, BVOP                    : Yes
Format settings, Matrix                  : Custom
Format settings, GOP                     : Variable
Codec ID                                 : V_MPEG2
Codec ID/Info                            : MPEG 1 or 2 Video
Duration                                 : 22 min 57 s
Bit rate mode                            : Variable
Bit rate                                 : 5 129 kb/s
Maximum bit rate                         : 9 800 kb/s
Width                                    : 720 pixels
Height                                   : 480 pixels
Display aspect ratio                     : 4:3
Frame rate mode                          : Variable
Frame rate                               : 24.503 FPS
Original frame rate                      : 29.970 (30000/1001) FPS
Standard                                 : Component
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Compression mode                         : Lossy
Bits/(Pixel*Frame)                       : 0.606
Time code of first frame                 : 00:59:58;00
Time code source                         : Group of pictures header
Stream size                              : 842 MiB (89%)
Language                                 : English
Default                                  : No
Forced                                   : No
Color primaries                          : BT.601 NTSC
Transfer characteristics                 : BT.601
Matrix coefficients                      : BT.601
Original source medium                   : DVD-Video

What would one do here? The video appears to be progressive when stepping through it. MediaInfo says it's *close* to 23.976. Do I let it force 23.976 and drop/dupe where needed? 

Since there's no telecine here, I take it I would not use Detelecine and just force the frame rate?

**Comment by donmelton** [comment]
@cameronks How does it look just using the defaults from `transcode-video`? It should add `--rate=23.976 --cfr` to the `HandBrakeCLI` options. And that should look fine. For most DVD content, there's no need to deinterlace or detelecine.

**Comment by cameronks** [comment]
I'll test it and see. 

`--rate=23.976 --cfr` is applied automatically, and is basically the same as `--force-rate 23.976` if applied manually in `transcode-video` correct?

**Comment by donmelton** [comment]
@cameronks Correct!

And sorry I took so long to respond, I was out walking the dog. :)

**Comment by cameronks** [comment]
Ok, so I've tested a variety of TV on DVD to get a better grasp of how to handle all the different kinds of material. It's way more complicated than encoding HD which is generally all progressive.

Example 1: Three's Company on DVD seems to be all Interlaced. Stepping through the material, each frame is unique and there's visible combing in almost all motion. For this, I simply used the Yadif deinterlacer with `-H deinterlace=default ` and left framerate at 29.97.

I know there is a method of using Bob and doubling the framerate, but I didn't care to keep the soap opera effect. 

Example 2: Friends on DVD seems to be Telecined. Every 4th and 5th frame is combed. But if I use `--filter detelecine` at 23.976, there is still some occasional combing at times, noticeably on some transitions. I added  `--filter comb-detect=permissive` which seems to catch these remaining areas without being too overzealous by decombing too many frames like the default settings seem to do in this case.

Example 3: The Simpsons on DVD has a variable frame rate on some seasons. If I run `--filter detelecine` at 23.976 it seems to fix the problem, but like Friends, there are occasional combed areas. Adding the same `--filter comb-detect=permissive` filter seems to fix this. BUT, alternatively, just running `--filter comb-detect=permissive` without  `--filter detelecine` also seems to fix the problem. I'm thinking of avoiding `--filter detelecine` in this case may be a better choice since the material really doesn't seem to have a consistent pattern.

Let me know if any of that doesn't sound right as I figure out what's the best way to handle these :)
